syncer/syncer.go

```diff
@@ -333,7 +334,17 @@ func (s *Syncer) Init(ctx context.Context) (err error) {
 	}
 
 	if s.cfg.OnlineDDL {
-		s.onlineDDL, err = onlineddl.NewRealOnlinePlugin(tctx, s.cfg)
+		shadowReg, err := regexp.Compile(fmt.Sprintf(`%s%s`,
```

Comment:
I think we need to support multiple expressions here to extract the right real table name